physionet-django/user/views.py

```diff
@@ -58,14 +60,18 @@
 
 @method_decorator(allow_post_during_maintenance, 'dispatch')
 class LoginView(auth_views.LoginView):
-    template_name = 'user/login.html'
-    extra_context = {
-        'enable_sso': settings.ENABLE_SSO,
-        'sso_login_button_text': settings.SSO_LOGIN_BUTTON_TEXT,
-    }
+    template_name = 'sso/login.html' if settings.ENABLE_SSO else 'user/login.html'
     authentication_form = forms.LoginForm
     redirect_authenticated_user = True
 
+    if settings.ENABLE_SSO:
```

Comment:
I'm wondering whether it's not a little cleaner to have this condition inside the function. Sort of like:
```python
def get_context_data(self, *args, **kwargs):
    context = super().get_context_data(*args, **kwargs)
    if not settings.ENABLE_SSO:
        return context
    sso_extra_context = {
        'sso_login_button_text': settings.SSO_LOGIN_BUTTON_TEXT,
        'sections': Section.objects.filter(page=Page.LOGIN_INST),
    }
    return {**context, **sso_extra_context}
```

Reply:
But maybe I'm wrong :) Definitely add the `super().get_context_data` call though